v2/auth/cachedauth/sign_in.go

```diff
@@ -0,0 +1,64 @@
+package cachedauth
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/SKF/go-utility/v2/auth"
+)
+
+var lock sync.RWMutex
+var tokens auth.Tokens
+
+var config *Config
+
+// Config is the configuration of the package
+type Config struct {
+	Stage string
+}
+
+// Configure will configure the package
+// please do not use go-utility/v2/auth together with this package
+func Configure(conf Config) {
+	lock.Lock()
+	defer lock.Unlock()
+
+	config = &conf
+
+	auth.Configure(auth.Config{Stage: conf.Stage})
+}
+
+// GetTokens will return the cached tokens
+//
+// note: Does not refresh the tokens
+func GetTokens() auth.Tokens {
+	lock.RLock()
+	defer lock.RUnlock()
+
+	return tokens
+}
+
+// SignIn is thread safe and only returns new tokens if the old tokens are about to expire
+func SignIn(ctx context.Context, username, password string) (err error) {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if config == nil {
+		return errors.New("cachedauth is not configured")
+	}
+
+	const tokenExpireDurationDiff = 5
```

Comment:
What unit is this? Isn't `IsTokenValid` supposed to take a `time.Duration` here?

Reply:
ops, this is Nanoseconds. Might be a bit optimistic ^^